project-koku/koku
# [COST-244]: Disable sources only through platform sources availability checker

Platform Sources routinely polls cost management for source status updates.  As part of this availability status check we perform the standard provider onboarding checks to ensure we are still able to retrieve cost and usage reports for the source.  In the event that a source fails these checks the provider will become deactivated and we will not make any subsequent ingest attempts until the user has corrected the configuration and the source status has passed.

On initial source creation, the provider object will not be created so there will be no ingest attempts made.
If the source was previously successfully created and the user misconfigures the cloud backend (deletes AWS bucket) then the next time the source is polled for status update the provider checks will fail and the provider becomes disabled.

This change removes the ProviderStatus object which was intended to serve a similar purpose.  There has been occasions in production where ProviderStatus disabled a source and the settings were actually correct.  Since there was no way to re-activate the source, these providers never ingested data.

**Testing**
1. Create an AWS Source
2. Modify the Cost and Usage report so that it fails our onboarding check
3. Hit the sources-status endpoint and verify source avaiability check fails and provider is disabled.
4. Verify unavaiabile status in sources-api
5. Verify provider is disabled
6. Attempt to ingest account with /download and verify that the account is skipped
7. Fix the AWS account by checking the "Include resource IDs" checkbox in the cost usage report
8. Hit the sources-status endpoint and verify source avaiability check fails and provider is enabled.
9. Attempt to ingest account with /download and verify that account ingest is attempted
10. Update the source with an invalid bucket and verify that the provider is not active.
11.  Correct the bucket and verify the provider becomes active again

**Test Results**
[active_provider_ut.txt](https://github.com/project-koku/koku/files/5059193/active_provider_ut.txt)
